utils/src/test/java/org/web3j/utils/NumericTest.java

```diff
@@ -72,14 +72,8 @@ public void testQuantityDecode() {
 
     @Test
     public void testQuantityDecodeLeadingZero() {
-        assertThrows(MessageDecodingException.class, () -> Numeric.decodeQuantity("0x0400"));
-        assertThrows(MessageDecodingException.class, () -> Numeric.decodeQuantity("0x001"));
-    }
-
-    @Test
-    public void testQuantityDecodeLeadingZeroException() {
-
-        assertThrows(MessageDecodingException.class, () -> Numeric.decodeQuantity("0x0400"));
+        assertEquals(Numeric.decodeQuantity("0x0400"), (BigInteger.valueOf(1024L)));
+        assertEquals(Numeric.decodeQuantity("0x001"), (BigInteger.valueOf(1L)));
```

Comment:
I would handle more corner cases. 
For example: 0x000

Reply:
yeah added one more corner case "0x00f"